src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp

```diff
@@ -0,0 +1,103 @@
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "simulation_defines.hpp"
+#include "particles/traits/GetAtomicNumbers.hpp"
+#include "traits/attribute/GetChargeState.hpp"
+#include "algorithms/math/floatMath/floatingPoint.tpp"
+#include "plugins/radiation/utilities.hpp"
+
+/** \file AlgorithmADK.hpp
+ *
+ * IONIZATION ALGORITHM for the ADK model
+ *
+ * - implements the calculation of ionization probability and changes charge states
+ *   by decreasing the number of bound electrons
+ * - is called with the IONIZATION MODEL, specifically by setting the flag in @see speciesDefinition.param */
+
+namespace picongpu
+{
+namespace particles
+{
+namespace ionization
+{
+
+    /** \struct AlgorithmADK
+     *
+     * \brief calculation for the Ammosov-Delone-Krainov tunneling model */
+    struct AlgorithmADK
+    {
+
+        /** Functor implementation
+         * \tparam EType type of electric field
+         * \tparam BType type of magnetic field
+         * \tparam ParticleType type of particle to be ionized
+         *
+         * \param bField magnetic field value at t=0
+         * \param eField electric field value at t=0
+         * \param parentIon particle instance to be ionized with position at t=0 and momentum at t=-1/2
+         */
+        template<typename EType, typename BType, typename ParticleType >
+        HDINLINE void
+        operator()( const BType bField, const EType eField, ParticleType& parentIon, float_X randNr )
+        {
+
+            const float_X protonNumber  = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
+            float_X chargeState         = attribute::getChargeState(parentIon);
+
+            uint32_t cs         = math::float2int_rd(chargeState);
+            const float_X pi    = precisionCast<float_X>(M_PI);
+            /* electric field in atomic units - only absolute value */
+            float_X eInAU       = math::abs(eField)*UNIT_EFIELD / SI::ATOMIC_UNIT_EFIELD;
+
+            /* effective principal quantum number */
+            float_X nEff        = protonNumber / math::sqrt(2*SI::IONIZATION_ENERGY[cs]/SI::ATOMIC_UNIT_ENERGY);
+            /* nameless variable for convenience dFromADK*/
+            float_X dBase       = 4 * util::cube(protonNumber) / (( eInAU ) * util::quad(nEff)) ;
+            float_X dFromADK    = math::pow(dBase,nEff);
+
+            /* ionization rate */
+            float_X rateADK     = math::sqrt(3 * util::cube(nEff) * eInAU / (pi * util::cube(protonNumber))) \
+                                    * eInAU * util::square(dFromADK) / (8 * pi * protonNumber) \
+                                    * math::exp(-2 * util::cube(protonNumber) / (3 * util::cube(nEff) * eInAU));
+
+            /* ionization probability
+             * calculated like a Poissonian induced process
+             */
+            float_X probADK     = rateADK * SI::DELTA_T_SI/SI::ATOMIC_UNIT_TIME;
```

Comment:
actually it is not dangerous
I have to multiply the rate by the time step to get a probability

when I change that according to normalization, we get:
``` 
probADK = rateADK * 1.0 / ATOMIC_UNIT_TIME
```
but here the time step is in units of `UNIT_TIME` and `ATOMIC_UNIT_TIME` is also in units of `UNIT_TIME`

if I optimize that by writing 
```
probADK = rateADK / ATOMIC_UNIT_TIME
```
this becomes **TOTALLY** misleading because it looks as if I were **dividing** the rate by the time. 

I could of course introduce new variables but that also seems a bit useless and superfluous